### PR TITLE
Added support for my new FeatherS2 Neo esp32s2 board

### PR DESCRIPTION
**Pull request by UnexpectedMaker**
As per the title, here are the files for my FeatherS2 Neo board.

### CODE REVIEW

File: ports/esp32s2/boards/unexpectedmaker_feathers2_neo/mpconfigboard.mk [hunk 1]
```diff
@@ -0,0 +1,20 @@
+USB_VID = 0x239A
+USB_PID = 0x80B5
```
Comment (tannewt) [inline ports/esp32s2/boards/unexpectedmaker_feathers2_neo/mpconfigboard.mk:2]: Why are you using this PID? I see it allocated to something else internally. Maybe request one from Espressif instead?

**Review by tannewt — CHANGES_REQUESTED** [review]

File: ports/esp32s2/boards/unexpectedmaker_feathers2_neo/mpconfigboard.mk (hunk 1 above)
Comment (tannewt) [reply ports/esp32s2/boards/unexpectedmaker_feathers2_neo/mpconfigboard.mk:2]: Looks like you did get the PID from Espressif. The PID is correct but the VID is wrong. It should be `0x303a` for Espressif.

Comment (tannewt) [reply ports/esp32s2/boards/unexpectedmaker_feathers2_neo/mpconfigboard.mk:2]: I'll fork this and fix.

**Comment by tannewt** [comment]
Replaced by #5294

File: ports/esp32s2/boards/unexpectedmaker_feathers2_neo/mpconfigboard.mk (hunk 1 above)
Comment (UnexpectedMaker) [reply ports/esp32s2/boards/unexpectedmaker_feathers2_neo/mpconfigboard.mk:2]: Super sorry! I totally forgot I got the FS2 PID from Adafruit, so only updated the PID, not the VID.
Thanks for taking care of this @tannewt !

**Pull request closed without merging**